zrml/prediction-markets/src/lib.rs

```diff
@@ -1716,6 +1721,82 @@ mod pallet {
             Ok(Some(weight).into())
         }
 
+        /// Allows the manual opening and closing for "broken" markets.
+        /// A market is "broken", if an unexpected chain stall happened
+        /// and the auto open / close was scheduled during this time.
+        ///
+        /// # Weight
+        ///
+        /// Complexity: `O(max{n, m})`,
+        /// where `n` is the number of market ids,
+        /// which open at the same time as the specified market,
+        /// and `m` is the number of market ids,
+        /// which close at the same time as the specified market.
+        #[pallet::call_index(21)]
+        #[pallet::weight((
+            T::WeightInfo::manually_open_market(CacheSize::get())
+                .max(T::WeightInfo::manually_close_market(CacheSize::get())),
+            Pays::Yes,
+        ))]
+        #[transactional]
+        pub fn manually_open_or_close_market(
+            origin: OriginFor<T>,
+            #[pallet::compact] market_id: MarketIdOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?;
+
+            let market = zrml_market_commons::Pallet::<T>::market(&market_id)?;
+            let now = zrml_market_commons::Pallet::<T>::now();
+            let range = match &market.period {
+                MarketPeriod::Block(_) => {
+                    return Err(Error::<T>::NotAllowedForBlockBasedMarkets.into());
+                }
+                MarketPeriod::Timestamp(ref range) => range,
+            };
+
+            let should_be_opened = range.start < now && now < range.end;
+            let should_be_closed = range.end <= now;
+            debug_assert!(!(should_be_opened && should_be_closed), "Both cases can not happen.");
+
+            let weight = if should_be_opened {
+                let range_start_time_frame = Self::calculate_time_frame_of_moment(range.start);
+                let open_ids_len = MarketIdsPerOpenTimeFrame::<T>::try_mutate(
+                    range_start_time_frame,
+                    |ids| -> Result<u32, DispatchError> {
+                        let ids_len = ids.len() as u32;
+                        let position = ids
+                            .iter()
+                            .position(|i| i == &market_id)
+                            .ok_or(Error::<T>::MarketNotInOpenTimeFrameList)?;
+                        let _ = ids.swap_remove(position);
+                        Ok(ids_len)
+                    },
+                )?;
+                Self::open_market(&market_id)?;
+                T::WeightInfo::manually_open_market(open_ids_len)
+            } else if should_be_closed {
```

Comment:
MarketIdsPerOpenTimeFrame is removed in this PR: https://github.com/zeitgeistpm/zeitgeist/pull/1197

I suggest to remove `should_be_opened` and only leave `should_be_closed`.

Reply:
Yeah, sounds good.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/1160/commits/5f107f09753a7ccf4ac85b12e1820dab780d4b86